lib/src/shared/main/java/com/couchbase/lite/Blob.java

```diff
@@ -458,16 +461,83 @@ public boolean equals(Object o) {
             : Arrays.equals(getContent(), m.getContent());
     }
 
+    @SuppressWarnings("NoFinalizer")
+    @SuppressFBWarnings("DE_MIGHT_IGNORE")
+    @Override
+    protected void finalize() throws Throwable {
+        if (blobContentStream != null) {
+            try { blobContentStream.close(); }
+            catch (IOException ignore) { }
+        }
+        super.finalize();
+    }
+
     //---------------------------------------------
     // Private (in class only)
     //---------------------------------------------
 
-    private Map<String, Object> jsonRepresentation() {
-        final Map<String, Object> json = new HashMap<>(getProperties());
-        json.put(META_PROP_TYPE, TYPE_BLOB);
-        if (blobDigest != null) { json.put(PROP_DIGEST, blobDigest); }
-        else { json.put(PROP_DATA, getContent()); }
-        return json;
+    @Nullable
+    private byte[] copyBytes(@Nullable byte[] b) {
+        if (b == null) { return null; }
```

Comment:
This method basically means that the user cannot modify the content of a blob unless they set it back again.  This seems to differ from both the existing behavior and others.  Was this a part of a discussion?

Reply:
You are correct.  Blobs are effectively immutable.
It was not discussed.  My bad.
On the other hand, I totally stand by this change.  Without it, the digest is probably a lie.

Reply:
On the third hand, now the problem of large in memory arrays is even worse since a copy is not only made on the way in but on the way out as well.

Reply:
It was not copied in either place when I started.  It is copied in both, now.
It is true that there will be two copies in memory, at some point.  It would take a very naive developer, indeed, to have 3.
In Java-land, passing things to a constructor is an idiom.  It means "I'm keeping a copy of this".  ... unless the "this" is also immutable.  It is such an idiom that I have had to tell the standard Java static analysis tools to ignore a "Probable bug".